src/compas/datastructures/mesh/mesh.py

```diff
@@ -2953,42 +2953,22 @@ def unify_cycles(self, root=None):
             The mesh is modified in place.
 
         """
-
-        def unify(node, nbr):
-            # find the common edge
-            for u, v in self.face_halfedges(nbr):
-                if u in self.face[node] and v in self.face[node]:
-                    # node and nbr have edge u-v in common
-                    i = self.face[node].index(u)
-                    j = self.face[node].index(v)
-                    if i == j - 1 or (j == 0 and u == self.face[node][-1]):
-                        # if the traversal of a neighboring halfedge
-                        # is in the same direction
-                        # flip the neighbor
-                        self.face[nbr][:] = self.face[nbr][::-1]
-                        return
-
-        if root is None:
-            root = self.face_sample(size=1)[0]
-
+        vertex_index = {vertex: index for index, vertex in enumerate(self.vertices())}
+        index_vertex = {index: vertex for index, vertex in enumerate(self.vertices())}
```

Comment:
Besides the nice terseness of this, wouldn't it be more efficient (and marginally more readable also) to do a single `for` loop enumerating over the vertices, and create the two indices in one go?  Something like

```suggestion
        vertex_index = {}
        index_vertex = {}
        for index, vertex in enumerate(self.vertices()):
                vertex_index[vertex] = index
                index_vertex[index] = vertex
```

Reply:
happy to make the change but not sure if it will be faster. comprehensions are highly optimised in recent python versions...

Reply:
ah ok, in that case, no idea what's faster